srml/session/src/lib.rs

```diff
@@ -134,18 +134,23 @@ pub trait ShouldEndSession<BlockNumber> {
 	fn should_end_session(now: BlockNumber) -> bool;
 }
 
+/// Ends the session after a fixed period of blocks.
+///
+/// The first session will have length of `Period + Offset`, and
+/// the following sessions will have length of `Period`.
 pub struct PeriodicSessions<
 	Period,
 	Offset,
 >(PhantomData<(Period, Offset)>);
 
 impl<
-	BlockNumber: Rem<Output=BlockNumber> + Saturating + Zero,
+	BlockNumber: Rem<Output=BlockNumber> + Sub<Output=BlockNumber> + Zero + PartialOrd,
 	Period: Get<BlockNumber>,
 	Offset: Get<BlockNumber>,
 > ShouldEndSession<BlockNumber> for PeriodicSessions<Period, Offset> {
 	fn should_end_session(now: BlockNumber) -> bool {
-		((now.saturating_sub(Offset::get())) % Period::get()).is_zero()
+		let offset = Offset::get();
+		now > offset && ((now - offset) % Period::get()).is_zero()
```

Comment:
nice, the previous impl would always succeed if `Offset::get() > now` because of `saturating_sub` right?

Reply:
right